spring-cloud-aws-sqs/src/main/java/io/awspring/cloud/sqs/invocation/EndpointMessageHandler.java

```diff
@@ -0,0 +1,281 @@
+		SqsListener sqsListenerAnnotation = AnnotationUtils.findAnnotation(method, SqsListener.class);
+		if (sqsListenerAnnotation != null && sqsListenerAnnotation.value().length > 0) {
+			Set<String> logicalEndpointNames = resolveDestinationNames(sqsListenerAnnotation.value());
+			return SqsEndpoint.from(logicalEndpointNames)
+					.factoryBeanName(resolveName(sqsListenerAnnotation.factory())[0])
+					.pollTimeoutSeconds(resolveInteger(sqsListenerAnnotation.pollTimeoutSeconds()))
+					.simultaneousPollsPerQueue(resolveInteger(sqsListenerAnnotation.concurrentPollsPerContainer()))
+					.minTimeToProcess(resolveInteger(sqsListenerAnnotation.minSecondsToProcess()))
+					.async(CompletionStage.class.isAssignableFrom(method.getReturnType()))
+					.queuesAttributes(logicalEndpointNames.stream()
+							.collect(Collectors.toMap(name -> name, this::getQueueAttributes)))
+					.build();
+		}
+
+		return null;
+	}
+
+	private Integer resolveInteger(String pollingTimeoutSeconds) {
+		String[] pollingStrings = resolveName(pollingTimeoutSeconds);
+		return pollingStrings == null || pollingStrings.length == 0 || !StringUtils.hasText(pollingStrings[0]) ? null
+				: Integer.parseInt(pollingStrings[0]);
+	}
+
+	private Set<String> resolveDestinationNames(String[] destinationNames) {
+		Set<String> result = new HashSet<>(destinationNames.length);
+
+		for (String destinationName : destinationNames) {
+			result.addAll(Arrays.asList(resolveName(destinationName)));
+		}
+
+		return result;
+	}
+
+	private String[] resolveName(String name) {
+
+		if (!(getApplicationContext() instanceof ConfigurableApplicationContext)) {
+			return wrapInStringArray(name);
+		}
+
+		ConfigurableApplicationContext applicationContext = (ConfigurableApplicationContext) getApplicationContext();
+		ConfigurableBeanFactory configurableBeanFactory = applicationContext.getBeanFactory();
+
+		String placeholdersResolved = configurableBeanFactory.resolveEmbeddedValue(name);
+		BeanExpressionResolver exprResolver = configurableBeanFactory.getBeanExpressionResolver();
+		if (exprResolver == null) {
+			return wrapInStringArray(name);
+		}
+		Object result = exprResolver.evaluate(placeholdersResolved,
+				new BeanExpressionContext(configurableBeanFactory, null));
+		if (result instanceof String[]) {
+			return (String[]) result;
+		}
+		else if (result != null) {
+			return wrapInStringArray(result);
+		}
+		else {
+			return wrapInStringArray(name);
+		}
+	}
+
+	private QueueAttributes getQueueAttributes(String queue) {
+		try {
+			logger.debug("Fetching queue attributes for queue " + queue);
+			String queueUrl = this.sqsAsyncClient.getQueueUrl(req -> req.queueName(queue)).get().queueUrl();
+			Map<QueueAttributeName, String> attributes = this.sqsAsyncClient
+						.getQueueAttributes(req -> req.queueUrl(queueUrl)).get().attributes();
+			boolean hasRedrivePolicy = attributes.containsKey(QueueAttributeName.REDRIVE_POLICY);
+			boolean isFifo = queue.endsWith(".fifo");
```

Comment:
As far as I remember there is an attribute in the queue which tells if it is fifo or not. I would rely on that instead of the name.

Reply:
Sure, I agree. I developed this PR using `LocalStack`, and I just got an empty set of attributes from there, so I left this logic from the previous version. I think we can change this when we add support for `FIFO` queues.